[Ng Hong Liang] iP

jinnhl opened the pull request:
# Jeff the task manager
> “My name is Jeff.” – Channing Tatum

Jeff allows you to have all your to-dos in one easy to access location. It's,
- text-based 📝 
- easy to use 👌 
- ~~fast~~ SUPER FAST to use 🔥 

How do I get Jeff? It's as easy as,
1. download it from here.
2. double-click it.
3. add your task.

And _**voilà**_, Jeff would do everything for you!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Reminders
- [ ] Working GUI (coming soon)

jinnhl committed:
Level 1. Greet,Echo,Exit

jinnhl committed:
Level2. Add,List

jinnhl committed:
Level 3. Mark as Done

jinnhl committed:
Level 4. ToDos,Events,Deadlines

jinnhl committed:
A-TextUiTesting done

jinnhl committed:
Polished the comments in Todo, Deadline and Event

jinnhl committed:
Level 5. Handle Erros

jinnhl committed:
Converted the storage type from Array to Arraylist + added delete function and it's exception handling

jinnhl committed:
Updated some testcases

jinnhl committed:
Added read and write feature

jinnhl committed:
Added better formatting for date and time

jinnhl committed:
Merge branch 'branch-Level-7'

jinnhl committed:
Merge branch 'branch-Level-8'

DannyDakota commented on src/main/java/TimeParse.java:
```diff
@@ -0,0 +1,25 @@
+        this.helper();
+    }
+
+    private LocalTime helper() {
```
maybe can use getTime here 😄

DannyDakota commented on src/main/java/DateParse.java:
```diff
@@ -0,0 +1,71 @@
+        this.helper();
+    }
+
+    private LocalDate helper() {
```
maybe can use getDate here? 😄

DannyDakota commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,89 @@
+     * @param curr Holds the capacity for the stored list, used to check if request is out of bounds
+     * @throws DukeException
+     */
+    public void checker(String[] arr, int curr) throws DukeException {
```
maybe can use a more descriptive name for the method 😄 
e.g
commandChecker
stringChecker
inputTester

DannyDakota commented on src/main/java/ReadFile.java:
```diff
@@ -0,0 +1,54 @@
+        while (s.hasNext()) {
+            String[] inputLine = s.nextLine().split("\\| ", 4);
+            Task curr;
+            switch(inputLine[0]) {
```
switch cases shouldn't be indented according to the coding standard 😄 
otherwise v nice use of switch to differentiate the various commands

DannyDakota reviewed (commented):
few suggestions to naming conventions and coding standards for switch cases. Otherwise, solid work handsome!

pyk595 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+    protected TimeParse time;
+
+    /**
+     * Constructor for Deadline class
```
Might be better to put a line after the JavaDoc descriptor and the params?

pyk595 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+import java.io.FileNotFoundException;
```
Can consider putting your Duke.java in a package?

pyk595 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+            // If user inputs bye, terminate the program
+            if (input.equals("bye")) {
+                end = true;
+            }
```
Good use of in-line comments, they adhere to the coding standard's requirement as well

Tiffanylin21 replied:
Agreed! I like how you use in-line comments to help in code readability and understanding the code :)

pyk595 commented on src/main/java/ReadFile.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class ReadFile {
+
+    public static final String ADDRESS = "data/duke.txt";
```
Good use of full caps for constants!

pyk595 commented on src/main/java/WriteFile.java:
```diff
@@ -0,0 +1,58 @@
+        fw.close();
+    }
+//
+//    public static void main(String[] args) {
```
Can consider removing these few lines of code if they are not used?

pyk595 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Good boolean variable naming!

pyk595 commented on src/main/java/ReadFile.java:
```diff
@@ -0,0 +1,54 @@
+                case "T " :
+                    curr = new Todo(inputLine[2]);
+                    break;
+                case "D " :
```
Same thing as mentioned by @DannyDakota, indentation should be removed

pyk595 commented on src/main/java/WriteFile.java:
```diff
@@ -0,0 +1,58 @@
+
+    public static final String ADDRESS = "data/duke.txt";
+
+    public WriteFile(ArrayList<Task> store) throws IOException {
```
Store might not be the best name for a task list, maybe naming it taskList might be a better choice?

pyk595 reviewed (commented):
Overall, most of your code adheres to the coding standard, good job! Just a few minor nitpicks here and there

Tiffanylin21 commented on src/main/java/DateParse.java:
```diff
@@ -0,0 +1,71 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.time.format.DateTimeFormatter;
```
Perhaps consider separating similar ones into blocks?

Tiffanylin21 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+        this.dateInfo = this.date.toString() + " " + this.time.toString();
+    }
+
+    public String whatType() {
```
Perhaps consider naming the method with verbs such as 'getType()'?

Tiffanylin21 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
-        System.out.println("Hello from\n" + logo);
+        Scanner scanner = new Scanner(System.in);
+        String input = "";
+        Boolean end = false;
```
Perhaps use a name that sounds more like a boolean? What about something like 'isEnd' or 'isExited'?

Tiffanylin21 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+        Scanner scanner = new Scanner(System.in);
+        String input = "";
+        Boolean end = false;
+        ArrayList<Task> store = new ArrayList(100);
```
Perhaps consider using a plural noun as the name of the ArrayList?

Tiffanylin21 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,104 @@
+        // While loop ends when user inputs bye
+        while(!end) {
+            input = scanner.nextLine();
+            String[] splitInput = input.split(" ", 2);
```
Perhaps consider using a plural noun as a variable name here?

Tiffanylin21 reviewed (commented):
Overall, I found some minor issues in naming. Otherwise, I really like the in-line comments in the code :)

jinnhl committed:
Add classes for A-MoreOOP

jinnhl committed:
Organize into Packages

jinnhl committed:
Add JUnit tests

jinnhl committed:
Add JavaDoc comments

jinnhl committed:
Merge branch 'branch-A-JavaDoc'

jinnhl committed:
Tweak the code to comply with a coding standard

jinnhl committed:
Merge branch 'branch-A-CodingStandard'

jinnhl committed:
Add more information to JavaDoc in command package

jinnhl committed:
Add find feature

jinnhl committed:
Merge branch 'branch-Level-9'

jinnhl committed:
Use Gradle

jinnhl committed:
Merge branch 'A-gradle'

jinnhl committed:
Add GUI

jinnhl committed:
Merge branch 'branch-Level-10'

jinnhl committed:
Add Assertions

Event and Deadline tasks does not check for assumptions that
string parsed as date and time are of correct size.

Event and Deadline tasks would cause Jeff to crash if certain
assumptions are not met.

Let's add assertions on those very variables have been added,
hence throwing an assertion failure instead of adding an
invalid entry into the list.

Assertions impact on performance is considered low compared
to its benefits.

jinnhl committed:
Improve Code Quality

Parser class have repeated conditional statements to check if the
user gave valid inputs.

To improve readability and code quality.

Let's refractor said conditional statements by
extract them into methods.

Easiest way to reuse code that have are duplicates of each other.
Allows easy navigation as they are on the same class file.

jinnhl committed:
Merge pull request #2 from jinnhl/branch-A-Assertions

Add Assertions

jinnhl committed:
Merge branch 'master' into branch-A-CodeQuality

jinnhl committed:
Merge pull request #3 from jinnhl/branch-A-CodeQuality

Improve Code Quality

jinnhl committed:
Add notes feature

jinnhl committed:
Merge branch 'branch-D-Notes'

jinnhl committed:
Improve GUI

jinnhl committed:
Add a User Guide

jinnhl committed:
Release v0.2
